pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -174,6 +176,49 @@ func (t ResolvedPipelineRunTask) IsStarted() bool {
 	return t.TaskRun != nil && t.TaskRun.Status.GetCondition(apis.ConditionSucceeded) != nil
 }
 
+// FirstAttemptStartTime returns the start time of the first time the ResolvedPipelineRunTask was attempted.
+// Returns nil if no attempt has been started.
+func (t *ResolvedPipelineRunTask) FirstAttemptStartTime(c clock.Clock) *metav1.Time {
```

Comment:
Do we want clock based funcs to be exposed to end-users, or is this an implementation detail that should be unexported?